app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -24,6 +24,7 @@ def initialize(ems, options = nil)
         @rgs               = resource_group_service(@config)
         @sas               = storage_account_service(@config)
         @mis               = managed_image_service(@config)
+        @vmis              = virtual_machine_image_service(@config, :location => @ems.provider_region)
```

Comment:
I believe it sets the location used by that instance of vmis, which I had to do because ```gather_data_for_this_region``` filters by location, and from what I could tell there wasn't a way to pass arguments to the ```list_all``` call that takes place inside ```gather_data_for_this_region```